coap/adapter.go

```diff
@@ -1,61 +1,145 @@
-package adapter
+package coap
 
 import (
-	"encoding/json"
-	"log"
+	"errors"
+	"fmt"
 	"net"
 
-	"github.com/dustin/go-coap"
+	mux "github.com/dereulenspiegel/coap-mux"
+	gocoap "github.com/dustin/go-coap"
+	"github.com/go-kit/kit/log"
 	"github.com/mainflux/mainflux"
 	broker "github.com/nats-io/go-nats"
-	"go.uber.org/zap"
 )
 
-const protocol string = "coap"
+// PubSub interface is used to access to NATS pub-sub model.
+type PubSub interface {
+	Publish(msg mainflux.RawMessage) error
```

Comment:
No need for naming parameters. Moreover, this will have to be updated once #188 is merged.